### PR TITLE
Fix warnings in xml activities and write some tests

### PR DESCRIPTION
**Pull request by Laniakea1999**

### CODE REVIEW

File: app/src/androidTest/java/com/github/polybooks/LoginGoogleTest.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    @Test
+    fun signInButtonGoogle() {
+        onView(withId(com.github.polybooks.R.id.sign_in_button)).perform(click())
+    }
+
+    @Test
+    fun signOutButtonGoogle() {
+        onView(withId(com.github.polybooks.R.id.tequila_log_button)).perform(click())
+    }
```
Comment (Huntsekkern) [inline app/src/androidTest/java/com/github/polybooks/LoginGoogleTest.kt:43]: I'm no espresso expert, but I believe those two tests can never fail as you click on a button, but doesn't assert any kind of result

File: app/src/main/res/layout/activity_login.xml [hunk 2]
```diff
@@ -36,7 +35,9 @@
         app:layout_constraintBottom_toTopOf="@+id/log_button_sub"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/textInfo2" />
+        app:layout_constraintTop_toBottomOf="@+id/textInfo2"
+        android:autofillHints="Password"
+        tools:ignore="LabelFor" />
```
Comment (Huntsekkern) [inline app/src/main/res/layout/activity_login.xml:40]: I don't think we want to ignore LabelFor on basic interface elements. They're necessary for accessibility. ignore="LabelFor" is here for stuff like logo or design elements.

File: app/src/androidTest/java/com/github/polybooks/LoginGoogleTest.kt (hunk 1 above)
Comment (Laniakea1999) [reply app/src/androidTest/java/com/github/polybooks/LoginGoogleTest.kt:43]: > I'm no espresso expert, but I believe those two tests can never fail as you click on a button, but doesn't assert any kind of result

They can fail if an error or exception is generated by google api, that's the only way to test the google authentification that I found so far.

File: app/src/main/res/layout/activity_login.xml [hunk 3]
```diff
@@ -51,7 +52,9 @@
         app:layout_constraintBottom_toTopOf="@+id/textInfo2"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/textInfo1" />
+        app:layout_constraintTop_toBottomOf="@+id/textInfo1"
+        android:autofillHints="Email"
+        tools:ignore="LabelFor" />
```
Comment (Huntsekkern) [inline app/src/main/res/layout/activity_login.xml:57]: Same here

File: app/src/main/res/layout/activity_signup.xml [hunk 4]
```diff
@@ -34,7 +34,9 @@
         android:inputType="textPassword"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/textInfo2" />
+        app:layout_constraintTop_toBottomOf="@+id/textInfo2"
+        android:autofillHints="Password"
+        tools:ignore="LabelFor" />
```
Comment (Huntsekkern) [inline app/src/main/res/layout/activity_signup.xml:39]: Same here?

File: app/src/main/res/layout/activity_signup.xml [hunk 5]
```diff
@@ -49,7 +51,9 @@
         app:layout_constraintBottom_toTopOf="@+id/textInfo2"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/textInfo1" />
+        app:layout_constraintTop_toBottomOf="@+id/textInfo1"
+        android:autofillHints=""
+        tools:ignore="LabelFor" />
```
Comment (Huntsekkern) [inline app/src/main/res/layout/activity_signup.xml:56]: Probably same here, but the ids are also poorly chosen, unless they're placeholders, so can't know for sure if we should ignore LabelFor or not

File: app/src/main/res/layout/activity_signup.xml [hunk 6]
```diff
@@ -167,7 +171,9 @@
         app:layout_constraintBottom_toTopOf="@+id/register_button"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/textInfo3" />
+        app:layout_constraintTop_toBottomOf="@+id/textInfo3"
+        android:autofillHints="Password"
+        tools:ignore="LabelFor" />
```
Comment (Huntsekkern) [inline app/src/main/res/layout/activity_signup.xml:176]: Same here

**Review by Huntsekkern — CHANGES_REQUESTED** [review]
Good on extracting the strings to resources!
Should stop ignoring LabelFor for approval.
And if possible make more relevant tests? (I know that it uses an external library, so it might not be so easy)

File: app/src/androidTest/java/com/github/polybooks/LoginGoogleTest.kt (hunk 1 above)
Comment (Huntsekkern) [reply app/src/androidTest/java/com/github/polybooks/LoginGoogleTest.kt:43]: Are you filling in data of user fields if successful? Or are you moving to the next activity?
If yes, those are things you could assert for.
If not, then, I don't see any way to do more than what you're already doing.

**Review by Huntsekkern — APPROVED** [review]
All good, thanks!

**Review by DGoedtkindtEPFL — APPROVED** [review]

**Pull request merged**